src/tutorial/hello-world/README.md

```diff
@@ -1,2 +1,245 @@
+
+The contract abi specification dictates that contract calls will append a 4 byte value (function selector) to their messages to determine which contract function the caller is intending to interact with. If you are curious as to what these look like you can find a registry of common 4byte function selectors in the [4 byte directory](https://www.4byte.directory/).
+
+**Modifying our contract to accept external function calls**
+To modify our contract to accept external calls for multiple functions we will add extract our `addTwo` logic into another macro. Then convert our `MAIN` macro into a function dispatcher.  
+```huff
+#define function addTwo(uint256,uint256) view returns(uint256)
+
+#define macro MAIN() = takes(0) returns(0) {
+    
+    // Get the function selector
+    0x00 
+    calldataload 
+    0xE0 
+    shr
+    
+    // Jump to the implementation of the ADD_TWO function if the calldata matches the function selector
+    __FUNC(addTwo) addTwo eq jumpi  
+
+    addTwo:
+        ADD_TWO()
+}
+
+#define macro ADD_TWO() = takes(0) returns(0) {
+    0x04 calldataload     // load first 32 bytes onto the stack - number 1
+    0x24 calldataload     // load second 32 bytes onto the stack - number 2
+    add                   // add number 1 and 2 and put the result onto the stack
+    
+    0x00 mstore           // place the result in memory
+    0x20 0x00 return      // return the result
+}
+```
+
+The first modifications we make will be within the ADD_TWO macro. On lines 1 and 2 we will shift the calldata offset by 4 bytes for both numbers, this is due to the 4 byte function selector that will be prepended to the calldata value. 
+
+Our `MAIN` macro has changed drastically. 
+The first 4 lines are concerned with isolating the function selector from the calldata.
+1. `0x00` pushed `[0]` onto the stack
+2. `calldataload` takes `[0]` as input and pushes the first 32 bytes of calldata onto the stack
+3. `0xE0` pushes `[224]` onto the stack. This magic number represents 256 bits - 32 bits (32 bytes - 4 bytes). 
+4. When followed by the shr this will place the function selector onto the stack
+
+The following lines will match the function selector on the stack and then jump to the code location where that code exists. Huff handles generating all jump logic for you.
+
+Under the hood the ADD_TWO() macro bytecode will be inlined of ADD_TWO() in the main macro. 
+
+Now you should be able to use libraries like ethers, or other contracts to call your huff contract! 
+
+We hope this now gives you a good understanding of the main concepts and all of the boiler plate you need to get started in huff!
+
+**Hello, world!**
```

Comment:
there is a lot of info so far on this page, especially if we add foundry tests.  may want to consider breaking this up into 2 pages?

Reply:
if we break it up i would propose stopping this page before line 118 where we refactor MAIN()

Reply:
I do agree its alot, however I think the way the headings are organised on the navigation bar makes it readible? Let me know what you think [here](https://deploy-preview-24--huffdocs.netlify.app/).